### PR TITLE
Use the plugin version defined by the Apache maven parent

### PR DESCRIPTION
**Pull request by CalvinKirs**
Use the plugin version defined by the Apache maven parent
Delete target field in the `java-doc` configuration

These are the upgraded versions:
```
maven-dependency-plugin 2.10 - 3.2.0
maven-assembly-plugin   3.1.0 - 3.3.0
maven-failsafe-plugin   2.22.0 - 2.22.2
maven-surefire-plugin   2.22.0 - 2.22.2
maven-jar-plugin        3.1.0 - 3.2.2
maven-enforcer-plugin   3.0.0-M2 - 3.0.0
maven-compiler-plugin 3.8.0 - 3.10.0
maven-resources-plugin 3.1.0 - 3.2.0
maven-source-plugin    3.0.1 - 3.2.1
```

- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Could you output those version to verify nothing changed actually?

**Comment by CalvinKirs** [comment]
> Could you output those version to verify nothing changed actually?

Thanks for the review, sorry I ignored it

These are the upgraded versions:
```
maven-dependency-plugin 2.10 - 3.2.0
maven-assembly-plugin   3.1.0 - 3.3.0
maven-failsafe-plugin   2.22.0 - 2.22.2
maven-surefire-plugin   2.22.0 - 2.22.2
maven-jar-plugin        3.1.0 - 3.2.2
maven-enforcer-plugin   3.0.0-M2 - 3.0.0
maven-compiler-plugin 3.8.0 - 3.10.0
maven-resources-plugin 3.1.0 - 3.2.0
maven-source-plugin    3.0.1 - 3.2.1

```

These are the unchanged versions:
```
maven-deploy-plugin     2.8.2 - 2.8.2
```

File: CHANGES.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Release Notes.
 * Upgrade protoc to 3.19.2.
 * Add Istio 1.13.1 to E2E test matrix for verification.
 * Upgrade Apache parent pom version to 25.
+* Use the plugin version defined by the Apache maven parent.
```
Comment (wu-sheng) [inline CHANGES.md:20]: Please be clear about all versions changed in the change logs.

Comment (CalvinKirs) [reply CHANGES.md:20]: Deeply thx. I thought plugins don't need to pay attention to these, thanks again for checking

Comment (wu-sheng) [reply CHANGES.md:20]: Thanks, all dependency(dev or runtime) things are sensitive to developers. So we care.

**Review by wu-sheng — APPROVED** [review]
LGTM @kezhenxu94 WDYT?

**Pull request merged**